src/bootstrap/lib.rs

```diff
@@ -1336,18 +1336,29 @@ impl Build {
     }
 
     fn beta_prerelease_version(&self) -> u32 {
+        fn extract_beta_rev<P: AsRef<Path>>(version_file: P) -> Option<String> {
+            let version = fs::read_to_string(version_file).ok()?;
+            let parts = version.splitn(2, "-beta.").collect::<Vec<_>>();
+            let count = parts.get(1).and_then(|s| s.find(' ').map(|p| (&s[..p]).to_string()));
+
+            count
+        }
+
         if let Some(s) = self.prerelease_version.get() {
             return s;
         }
 
-        // Figure out how many merge commits happened since we branched off master.
-        // That's our beta number!
-        // (Note that we use a `..` range, not the `...` symmetric difference.)
-        let count =
+        // First check if there is a version file available.
+        // If available, we read the beta revision from that file.
```

Comment:
Please leave a comment here that this only happens when building from the source tarball, where git is unavailable.

Reply:
Done. Added a sentence explaining when this condition could happen.